File: examples/java/greetings_project/greeting_lib/src/com/example/Greeting.java
package com.example;

import java.util.ArrayList;
import java.util.List;

public class Greeting {
    public static List<String> getGreetings(String greetingWord, List<String> names) {
        List <String> result = new ArrayList<>();
        for (String name : names) {
            result.add(String.format("%s %s!", greetingWord, name));
        }
       return result;
    }
}

File: examples/kotlin/simple_project/main/lib/Greeting.kt
package main.lib

class Greeting {
  companion object {
    private fun getGreeting(name: String): String {
      return "Hello $name!"
    }

    fun getGreetings(names: Array<String>): Array<String> {
      return Array(names.size) { i -> getGreeting(names.get(i)) }
    }
  }
}
